### PR TITLE
Feature/us 892 setup service summary component

### PR DESCRIPTION
**Pull request by anoveskey**
Tried to match the wireframes on this, but it may need some fine tuning based on peer feedback

### CODE REVIEW

**Comment by JeremySparkman** [comment]
A lot to go through but from what I've seen it looks good.

Only a couple small issues I found with Typescript. I seen that a couple of the components didn't have types for the parameters or return values. Here's an example of what I think should be added:

```import React, { FC, ReactElement } from 'react';

import { IProduct } from 'types/serviceRequest';

interface IProductListProps {
  products: IProduct[] | null;
}

const ProductList: FC<IProductListProps> = ({ products }: IProductListProps): null | ReactElement => {
  if (!products) {
    return null;
  }

  const productList = products.map(
    (product: IProduct, index: number): ReactElement => {
      return <span key={`product-${index}`}>{(index ? ', ' : '') + product.productType}</span>;
    },
  );

  return <span>{productList}</span>;
};

export default ProductList;``` 

Also ServiceSummary is just missing the Return type as well.

**Comment by JeremySparkman** [comment]
Also I ran the tests and I'm seeing a lot of issues with 
```
const tree: ReactElement 
```
It looks like it needs to be this instead:
```
const tree: null | ReactTestRendererJSON
```

File: app/components/ServiceSummary/index.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+}
+
+const ServiceSummary: React.FC<IServiceSummaryProps> = (props: IServiceSummaryProps): React.ReactElement => {
+  const { accountNumber, address, products } = props.serviceRequest;
+  const { city, line1, line2, state, zip } = address;
+
+  return (
+    <div className={'dte-wismo-service-summary'}>
+      <div className={'service-summary-section'}>
+        <span className={'service-summary-header'}>
```
Comment (JeremySparkman) [inline app/components/ServiceSummary/index.tsx:24]: I think we should use a more semantically correct html element for the summary headers

File: app/components/ServiceSummary/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        <FormattedMessage {...messages.addressLabel}/>
+      </span>
+        <Address city={city} line1={line1} line2={line2} state={state} zip={zip}/>
+      </div>
+      <div className={'service-summary-section'}>
+      <span className={'service-summary-header'}>
+        <FormattedMessage{...messages.accountNumberLabel}/>
+      </span>
+        <br/>
```
Comment (JeremySparkman) [inline app/components/ServiceSummary/index.tsx:33]: Also we should use CSS to create space instead of a <br/> unless it's needed for semantics such as with an Address

File: app/components/ProductList/getProductName.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+  switch (productCode) {
+    case 'IAC':
+      return 'Interuptable Air Conditioning';
+    case 'RES_GAS':
+      return 'Residential Gas Heating';
+    case 'RES_ELECTRIC':
+      return 'Residential Electric Service';
+    default:
+      return;
+  }
```
Comment (jmiller-nexient) [inline app/components/ProductList/getProductName.ts:11]: These names need to be internationalized.  The code value should map to the ID value in messages.ts.  You should be able to resolve the correct message to use with FormattedMessage like messages[code].  I think we did this for something with Covetrus but can't remember off the top of my head where.

File: app/components/ProductList/index.tsx [hunk 4]
```diff
@@ -1,18 +1,23 @@
...
   const { products } = props;
 
   if (!products) {
     return null;
   }
 
   const productList = products.map((product, index) => {
-    return <span key={`product-${index}`}>{ (index ? ', ' : '') + product}</span>;
+    const productName = getProductName(product.productType);
+    return <span key={`product-${index}`}>{ (index ? ', ' : '') + productName}</span>;
```
Comment (jmiller-nexient) [inline app/components/ProductList/index.tsx:20]: Per our coding standards (https://nexient.atlassian.net/wiki/spaces/DTE/pages/1040581086/MIMO+React+Coding+Guidelines), always put a blank line before return statements (unless they are the only statement in the block).

Comment (jmiller-nexient) [reply app/components/ProductList/index.tsx:20]: And looking at the snapshot, this will result in cleaner HTML.

File: app/components/ProductList/tests/index.test.tsx [hunk 5]
```diff
@@ -1,25 +1,36 @@
...
 
     expect(tree).toMatchSnapshot();
   });
 
-  it('should render correctly with an empty product list', () => {
-    const tree: ReactElement = renderer.create(
+  it('should render correctly when product list is empty', () => {
+    const tree = renderer.create(
       <ProductList products={null} />,
     ).toJSON();
```
Comment (jmiller-nexient) [inline app/components/ProductList/tests/index.test.tsx:36]: I suggest writing four tests:
1. ...when the product list contains multiple items
2. ...when the product list contains one item
3. ...when the product list is empty
4. ...when the product list is null

File: app/components/ServiceSummary/index.less [hunk 6]
```diff
@@ -0,0 +1,31 @@
+@import '../../styles/themes/index.less';
+
+@service-summary-prefix-cls: ~'@{dte-prefix}-service-summary';
+
+.@{service-summary-prefix-cls} {
+  border: 1px solid #b1b3b3;
```
Comment (jmiller-nexient) [inline app/components/ServiceSummary/index.less:6]: Always use the colors from the theme!

File: app/components/ServiceSummary/index.tsx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+  const { city, line1, line2, state, zip } = address;
+
+  return (
+    <div className={'dte-wismo-service-summary'}>
+      <div className={'service-summary-section'}>
+        <span className={'service-summary-header'}>
+        <FormattedMessage {...messages.addressLabel}/>
+      </span>
+        <Address city={city} line1={line1} line2={line2} state={state} zip={zip}/>
+      </div>
```
Comment (jmiller-nexient) [inline app/components/ServiceSummary/index.tsx:28]: If you review the architecture doc in Confluence, I called out separate components for each of the fields:
- AddressField
- AccountNumberField
- ProductListField

These fields encapsulate the Label (FormattedMessage) and value for each of the sections.  I would expect to see the following:

A base FormField component that either takes the MessageDescriptor or ReactNode for the `label` prop plus a children property of type ReactNode. This component simply lays out the structure for the label and value.

AddressFormField renders FormField with the appropriate label and passes Address as the child component.
AccountNumberFormField does the same for the account number
ProductNamesFormField does the same for the product names

Then all you have in the ServiceSummary is the layout for the three form fields.

File: app/types/serviceRequest.ts [hunk 8]
```diff
@@ -0,0 +1,64 @@
+export interface IAddress {
+  readonly city: string;
+  readonly line1: string;
+  readonly line2: (string | null);
+  readonly state: string;
+  readonly zip: string;
+}
+
+export interface IProduct {
```
Comment (jmiller-nexient) [inline app/types/serviceRequest.ts:9]: Every interface goes into a separate file.

**Comment by GaberMowiena** [comment]
One massive PR - Great Work though!

File: app/components/AccountNumberField/index.tsx [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+
+interface IAccountNumberFieldProps {
+  children: ReactNode;
+  label: MessageDescriptor;
+}
+
+const AccountNumberField: React.FC<IAccountNumberFieldProps> = (props: IAccountNumberFieldProps): React.ReactElement => {
+  const { children, label } = props;
+
+  return <BaseFormField children={children} label={label}/>;
```
Comment (jmiller-nexient) [inline app/components/AccountNumberField/index.tsx:14]: Children should not be called out as a discrete prop.  Children are passed as the inner elements of the component:
```
return (
  <BaseFormField label={label}>
    {children}
  </BaseFormField>
);
```

File: app/components/AccountNumberField/index.tsx [hunk 10]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from 'react';
+import { MessageDescriptor } from 'react-intl';
+
+import BaseFormField from 'components/BaseField';
+
+interface IAccountNumberFieldProps {
+  children: ReactNode;
+  label: MessageDescriptor;
```
Comment (jmiller-nexient) [inline app/components/AccountNumberField/index.tsx:8]: Why are these being passed in here?  This component provides no value and does nothing that the base component does.  The whole point of this component was that it passes the correct label text and child component (AccountNumber, in this case) to the base component.  The only prop that should be passed into this component is the account number!

File: app/components/AddressField/index.tsx [hunk 11]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from 'react';
+import { MessageDescriptor } from 'react-intl';
+
+import BaseFormField from 'components/BaseField';
+
+interface IAddressFieldProps {
+  children: ReactNode;
+  label: MessageDescriptor;
+}
```
Comment (jmiller-nexient) [inline app/components/AddressField/index.tsx:9]: Same issue here! This component should take the address in its props and pass the correct label text and child component (Address) to the base component.

File: app/components/BaseField/index.less [hunk 12]
```diff
@@ -0,0 +1,19 @@
+@import '../../styles/themes/index.less';
+
+.dte-wismo-base-field-section {
```
Comment (jmiller-nexient) [inline app/components/BaseField/index.less:3]: Follow the same structure used in all of the other LESS files in the app (and with Covetrus).  Declare the prefix at the top and use proper LESS syntax to build these styles.

File: app/components/OrderNumber/index.tsx [hunk 13]
```diff
@@ -11,7 +11,8 @@ const OrderNumber: React.FC<IOrderNumberProps> = (props: IOrderNumberProps): Rea
     return `${input.substr(0,4)} ${input.substr(4,4)} ${input.substr(8,3)} ${input.substr(11,1)}`;
   };
 
-  if (value.match(new RegExp('([0-9]{12})\d', 'g'))) {
+  const regex = new RegExp('^[0-9]{12}$', 'g');
+  if (!regex.test(value)) {
```
Comment (jmiller-nexient) [inline app/components/OrderNumber/index.tsx:15]: Always put a blank line before control statements like "if".  Same rules as we had for Covetrus.

File: app/components/ProductList/index.tsx [hunk 14]
```diff
@@ -1,25 +1,32 @@
...
+      </span>
+    );
   });
 
-  return (
-    <span>
-      {productList}
-    </span>
-  );
+  return <span>{renderProductNames}</span>;
```
Comment (jmiller-nexient) [inline app/components/ProductList/index.tsx:29]: Do we need the <span>?  Just use the Fragment syntax:
```
return (
  <>
    {renderProductNames()}
  </>
);
```

File: app/components/ProductList/index.tsx [hunk 15]
```diff
@@ -1,25 +1,32 @@
...
+const ProductList: React.FC<IProductListProps> = (props: IProductListProps): null | React.ReactElement => {
   const { products } = props;
 
   if (!products) {
     return null;
   }
 
-  const productList = products.map((product, index) => {
-    return <span key={`product-${index}`}>{ (index ? ', ' : '') + product}</span>;
+  const renderProductNames = products.map((product, index) => {
```
Comment (jmiller-nexient) [inline app/components/ProductList/index.tsx:19]: The name is a verb so it must be a function.

File: app/components/ProductList/tests/__snapshots__/index.test.tsx.snap [hunk 16]
```diff
@@ -1,17 +1,27 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
-exports[`<ProductList /> should render correctly 1`] = `
+exports[`<ProductList /> should render correctly when page loads 1`] = `
```
Comment (jmiller-nexient) [inline app/components/ProductList/tests/__snapshots__/index.test.tsx.snap:3]: Invalid "when" clause as this doesn't tell me anything and could be incorrect.  Fact is the other tests could be showing how the component renders when the page loads given those conditions.  So call out the conditions under which this result is expected.

File: app/components/ProductsListField/index.tsx [hunk 17]
```diff
@@ -0,0 +1,17 @@
+import React, { ReactNode } from 'react';
+import { MessageDescriptor } from 'react-intl';
+
+import BaseFormField from 'components/BaseField';
+
+interface IProductsListFieldProps {
+  children: ReactNode;
+  label: MessageDescriptor;
+}
```
Comment (jmiller-nexient) [inline app/components/ProductsListField/index.tsx:9]: Same issue here as the other two components! This component should take the array of product types in its props and pass the correct label text and child component (ProductList) to the base component.

File: app/components/ProductsListField/tests/index.test.tsx [hunk 18]
```diff
@@ -0,0 +1,39 @@
...
+  {
+    productType: 'RES_GAS',
+    serviceDate: '04-20-2020'
+  }
+];
+
+const fillerComponent = <ProductList products={fakeProducts}/>;
+
+describe('<ProductsListField />', () => {
+  it('should render correctly when page loads', () => {
```
Comment (jmiller-nexient) [inline app/components/ProductsListField/tests/index.test.tsx:30]: Same comment here (and everywhere else) about this test name.  "page loads" does not tell the viewer anything about this test.

File: app/components/ServiceSummary/index.less [hunk 19]
```diff
@@ -0,0 +1,31 @@
...
+  padding: 10px;
+  -webkit-column-count: 2;
+  @media screen and (max-width: 400px) {
+    column-count: 1;
+    -moz-column-count: 1;
+    -webkit-column-count: 1;
+  }
+}
+
+.service-summary-header {
```
Comment (jmiller-nexient) [inline app/components/ServiceSummary/index.less:19]: Invalid class name.  Use proper LESS syntax and build the class name up from the prefix above.

File: app/components/ServiceSummary/index.less [hunk 20]
```diff
@@ -0,0 +1,31 @@
...
+  font-size: @dte-wismo-font-size-2;
+  font-weight: @dte-wismo-font-weight-semi-bold;
+}
+
+.service-summary-section {
```
Comment (jmiller-nexient) [inline app/components/ServiceSummary/index.less:24]: Same here

File: app/components/ServiceSummary/index.tsx [hunk 21]
```diff
@@ -0,0 +1,39 @@
...
+        label={messages.addressLabel}
+      />
+      <AccountNumberFormField
+        children={<AccountNumber value={accountNumber}/>}
+        label={messages.accountNumberLabel}
+      />
+      <ProductsFormField
+        children={<ProductList products={products}/>}
+        label={messages.productListLabel}/>
+    </div>
```
Comment (jmiller-nexient) [inline app/components/ServiceSummary/index.tsx:35]: I expect this to be simply:
```
    <div className={'dte-wismo-service-summary'}>
      <AddressFormField address={address} />
      <AccountNumberFormField accountNumber={accountNumber} />
      <ProductsFormField products={products} />
    </div>
```

However, I am curious how this is displaying the fields in two columns on wide-screen.

File: app/components/TestPage/index.tsx [hunk 22]
```diff
@@ -127,6 +128,10 @@ const TestPage: React.FC = (): React.ReactElement => (
         </button>
       </p>
     </div>
+    <div>
+      <h4>Service Summary component:</h4>
+      <ServiceSummary serviceRequest={fakeServiceRequest}/>
+    </div>
```
Comment (jmiller-nexient) [inline app/components/TestPage/index.tsx:134]: You don't need to add this to the test page since it is on the OrderStatusPage - or at least, that's where it's supposed to be!

File: app/types/product.ts [hunk 23]
```diff
@@ -0,0 +1,4 @@
+export default interface IProduct {
+  readonly productType: (string | null);
+  readonly serviceDate: (string | null);
```
Comment (jmiller-nexient) [inline app/types/product.ts:3]: Neither of these are nullable.

File: app/types/serviceOrderEvent.ts [hunk 24]
```diff
@@ -0,0 +1,18 @@
+import IProduct from './product';
+import IServiceOrderEventStatusUpdate from './serviceOrderEventStatusUpdate';
+
+export default interface IServiceOrderEvent {
+  readonly appointment: {
```
Comment (jmiller-nexient) [inline app/types/serviceOrderEvent.ts:5]: This should be an IServiceAppointment interface.

File: app/types/serviceOrderEvent.ts [hunk 25]
```diff
@@ -0,0 +1,18 @@
...
+    readonly appointmentWindow: {
```
Comment (jmiller-nexient) [inline app/types/serviceOrderEvent.ts:6]: This should be an IServiceAppointmentWindow interface

File: app/types/serviceOrderEvent.ts [hunk 26]
```diff
@@ -0,0 +1,18 @@
...
+    }
+    readonly date: (string | null);
+  };
+  readonly contactPhoneNumber: (string | null);
+  readonly createdAt: (string | null);
+  readonly products: IProduct[];
+  readonly serviceOrderEventStatusUpdates: IServiceOrderEventStatusUpdate[];
+  readonly serviceOrderId: (string | null);
+  readonly status: (string | null);
+}
```
Comment (jmiller-nexient) [inline app/types/serviceOrderEvent.ts:18]: Look at the Open API Spec document in AzDO.  Anything listed as required there, is not nullable here.

File: app/types/serviceOrderEventStatusUpdate.ts [hunk 27]
```diff
@@ -0,0 +1,4 @@
+export default interface IServiceOrderEventStatusUpdate {
+  readonly createdAt: (string | null);
+  readonly serviceOrderEventStatus: (string | null);
+}
```
Comment (jmiller-nexient) [inline app/types/serviceOrderEventStatusUpdate.ts:4]: Same comment about looking at the Contract in AzDO and making this match.

File: app/types/serviceRequest.ts [hunk 28]
```diff
@@ -0,0 +1,18 @@
...
+  readonly address: IAddress;
+  readonly advisements: (string | null)[];
+  readonly serviceRequestStatusUpdates: IServiceRequestStatusUpdate[];
+  readonly premiseId: (string | null);
+  readonly products: IProduct[];
+  readonly serviceOrderEvents: IServiceOrderEvent[];
+  readonly serviceRequestType: (string | null);
+  readonly verificationEvents: IVerificationEvent[];
+  readonly wantDate: (string | null);
+}
```
Comment (jmiller-nexient) [inline app/types/serviceRequest.ts:18]: Again, look at the Contract in AzDO and making this match.  You have fields here that are actually a part of the IOrder interface, not IServiceRequest.

File: app/types/verificationEvent.ts [hunk 29]
```diff
@@ -0,0 +1,9 @@
+import IVerificationEventStatusUpdate from './verificationEventStatusUpdate';
+
+export default interface IVerificationEvent {
+  readonly createdAt: (string | null);
+  readonly verificationCode: (string | null);
+  readonly verificationDescription: (string | null);
+  readonly verificationEventStatus: (string | null);
+  readonly verificationEventStatusUpdates: IVerificationEventStatusUpdate[];
+}
```
Comment (jmiller-nexient) [inline app/types/verificationEvent.ts:9]: One more reminder about looking at the Contract in AzDO and making this match.

File: app/types/serviceRequestStatusUpdate.ts [hunk 30]
```diff
@@ -0,0 +1,4 @@
+export default interface IServiceRequestStatusUpdate {
+  readonly createdAt: (string | null);
+  readonly serviceRequestStatus: (string | null);
+}
```
Comment (jmiller-nexient) [inline app/types/serviceRequestStatusUpdate.ts:4]: Look at the API contract.

File: app/types/verificationEventStatusUpdate.ts [hunk 31]
```diff
@@ -0,0 +1,4 @@
+export default interface IVerificationEventStatusUpdate {
+  readonly createdAt: (string | null);
+  readonly verificationEventStatus: (string | null);
+}
```
Comment (jmiller-nexient) [inline app/types/verificationEventStatusUpdate.ts:4]: Here as well.

File: app/components/OrderNumber/index.tsx (hunk 13 above)
Comment (anoveskey) [reply app/components/OrderNumber/index.tsx:16]: I'm not sure why this change is being reflected. I put a space between the const and the if statement in my branch. I also tried doing another git add specifically on this file. Maybe github is just acting screwy...

File: app/components/AccountNumberField/tests/index.test.tsx [hunk 32]
```diff
@@ -0,0 +1,21 @@
...
+import { IntlProvider } from 'react-intl';
+import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
+import { translationMessages } from '../../../i18n';
+
+import AccountNumberField from 'components/AccountNumberField';
+
+const accountNumber = '111233455567';
+
+describe('<AccountNumberField />', () => {
+  it('should render correctly when page loads', () => {
```
Comment (jmiller-nexient) [inline app/components/AccountNumberField/tests/index.test.tsx:12]: I'm not going to hold up merging over this but, as I pointed out previously, this when clause is invalid.  Are you saying that it only renders correctly when the page loads?  It seems to me it should render this way all the time.

File: app/components/AccountNumberField/tests/index.test.tsx [hunk 33]
```diff
@@ -0,0 +1,21 @@
...
+    const tree: ReactTestRendererJSON | null = renderer.create(
```
Comment (jmiller-nexient) [inline app/components/AccountNumberField/tests/index.test.tsx:13]: Look at the PR that was merged this morning fixing all of these issues.  This is not the correct type to use.

File: app/components/AddressField/tests/index.test.tsx [hunk 34]
```diff
@@ -0,0 +1,28 @@
...
+  line1: '1234 That Street',
+  line2: null,
+  city: 'Anytown',
+  state: 'MI',
+  zip: '12345'
+};
+
+describe('<AddressField />', () => {
+  it('should render correctly when page loads', () => {
+    const tree: ReactTestRendererJSON | null = renderer.create(
```
Comment (jmiller-nexient) [inline app/components/AddressField/tests/index.test.tsx:20]: All the same comments here.

File: app/components/BaseField/index.less [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+.@{base-field-prefix-cls}-label {
+  font-size: @dte-wismo-font-size-2;
+  font-weight: @dte-wismo-font-weight-semi-bold;
+}
+
+.@{base-field-prefix-cls}-children {
+  color: @dte-gray-500;
+}
```
Comment (jmiller-nexient) [inline app/components/BaseField/index.less:21]: Ugh.  Nest the classes!
```
.@{base-field-prefix-cls} {
  &-section {
    display: inline-block;
    margin: 15px;
    width: 50%;

    @media screen and (max-width: 400px) {
      width: 100%;
    }

  &-label {
    font-size: @dte-wismo-font-size-2;
    font-weight: @dte-wismo-font-weight-semi-bold;
  }

  &-children {
    color: @dte-gray-500;
  }
}
```

Comment (jmiller-nexient) [reply app/components/BaseField/index.less:21]: The alternative is to do what Jeremy has been doing:
```
.@{base-field-prefix-cls} {
  &-section {
    display: inline-block;
    margin: 15px;
    width: 50%;

    @media screen and (max-width: 400px) {
      width: 100%;
    }

    .label {
      font-size: @dte-wismo-font-size-2;
      font-weight: @dte-wismo-font-weight-semi-bold;
    }

    .children {
      color: @dte-gray-500;
    }
  }
}
```
Then you just put "label" and "children" as the class names for the inner divs.

File: app/components/BaseField/index.tsx [hunk 36]
```diff
@@ -0,0 +1,28 @@
...
+
+  return (
+    <div className={'dte-wismo-base-field-section'}>
+      <label className={'dte-wismo-base-field-label'}>
+        <FormattedMessage {...label} />
+      </label>
+      <div>
+      <span className={'dte-wismo-base-field-children'}>
+        {React.Children.toArray(props.children)}
+      </span>
```
Comment (jmiller-nexient) [inline app/components/BaseField/index.tsx:22]: Prettier should have balked at the indents here.  The &lt;span> needs to be in two more spaces.

File: app/components/ProductList/index.tsx [hunk 37]
```diff
@@ -1,24 +1,35 @@
...
   const { products } = props;
 
   if (!products) {
     return null;
   }
 
   const productList = products.map((product, index) => {
-    return <span key={`product-${index}`}>{ (index ? ', ' : '') + product}</span>;
+    return (
+      // tslint:disable-next-line:jsx-key
```
Comment (jmiller-nexient) [inline app/components/ProductList/index.tsx:21]: Why is this here?  My guess is you need to remove the Fragment (&lt;>&lt;/>) and just return the FormattedMessage.

File: app/components/ProductList/tests/index.test.tsx [hunk 38]
```diff
@@ -1,28 +1,53 @@
 import React from 'react';
 import renderer from 'react-test-renderer';
 
-import TestRendererJsonType from 'types/TestRendererJsonType';
+import IProduct from 'types/product';
 
 import ProductList from '..';
+import TestRendererJsonType from '../../../types/TestRendererJsonType';
```
Comment (jmiller-nexient) [inline app/components/ProductList/tests/index.test.tsx:7]: Relative path?

File: app/components/ProductList/index.tsx (hunk 37 above)
Comment (anoveskey) [reply app/components/ProductList/index.tsx:21]: It's there because the return statement returns not just the FormattedMessage, but also the ", " in between each message. The only other way to properly return the list of formatted messages would be to apply the key attribute directly to the fragment like so:

`<React.Fragment key={product.productType}>
        <FormattedMessage {...messages[`${product.productType}`]} />
        {index < products.length - 1 ? ', ' : null}
      </React.Fragment>`
